plugins/pepkong/kong/plugins/pepkong/handler.lua

```diff
@@ -1,15 +1,102 @@
 local BasePlugin = require "kong.plugins.base_plugin"
-local access = require "kong.plugins.pepkong.pepmain"
+local jwt_decoder = require "kong.plugins.jwt.jwt_parser"
+local http = require "socket.http"
 
-local PEPPlugin = BasePlugin:extend()
+local build_form_params = require("kong.plugins.pepkong.utils").build_form_params
+
+local re_gmatch = ngx.re.gmatch
+
+local pepKongHandler = BasePlugin:extend()
+
+function pepKongHandler:new()
+    pepKongHandler.super.new(self, "pepkong")
+end
+
+local function retrieve_token()
+
+    local authorization_header = kong.request.get_header("authorization")
+    if authorization_header then
+        local iterator, iter_err = re_gmatch(authorization_header, "\\s*[Bb]earer\\s+(.+)")
+        if not iterator then
+            return nil, iter_err
+        end
+
+        local m, err = iterator()
+        if err then
+            return nil, err
+        end
+
+        if m and #m > 0 then
+            return m[1]
+        end
+    end
+end
+
+-- Use Keycloak authorization services to make authorization decision
+local function do_authorization(conf)
+
+    -- Retrieve token
+    local token, err = retrieve_token()
+    if err then
+        kong.log.err(err)
+        return kong.response.exit(500, {
+            message = "An unexpected error occurred"
+        })
+    end
```

Comment:
It seems to me that the error occurs only if the user does not send the JWT token correctly by header, am I right? in this case, I believe it is an error of the 4xx family, that is, the client made the request incorrectly.

In this case, I would return the http status 401 "Unauthorized", see [this](https://github.com/dojot/dojot/blob/development/subscription-engine/kafka-ws/app/routes/TicketRouter.js#L25) example.

Reply:
It happens when some error occurs during JWT token recovery process. The scenario where the user does not send the JWT token had not been done, but now is ok, check the last commit